caas/kubernetes/clientconfig/k8s.go

```diff
@@ -132,21 +132,18 @@ func contextsFromConfig(config *clientcmdapi.Config) (map[string]Context, error)
 }
 
 func cloudsFromConfig(config *clientcmdapi.Config, cloudName string) (map[string]CloudConfig, error) {
-
 	clusterToCloud := func(cluster *clientcmdapi.Cluster) (CloudConfig, error) {
 		attrs := map[string]interface{}{}
 
-		// TODO(axw) if the CA cert is specified by path, then we
-		// should just store the path in the cloud definition, and
-		// rely on cloud finalization to read it at time of use.
+		k8sCAData := cluster.CertificateAuthorityData
 		if cluster.CertificateAuthority != "" {
 			caData, err := ioutil.ReadFile(cluster.CertificateAuthority)
 			if err != nil {
 				return CloudConfig{}, errors.Trace(err)
 			}
-			cluster.CertificateAuthorityData = caData
+			k8sCAData = caData
```

Comment:
Trying to dig through what certificate-authority-data is supposed to be, I came across this:
https://insujang.github.io/2019-12-18/kubernetes-authentication/

Which seems to say that -data is supposed to be base64 encoding of .../ca.crt? This seems to assume that it is the direct data.

This post:
https://codefarm.me/2019/02/01/access-kubernetes-api-with-client-certificates/

also says to generate the content:
```
# Generate the <CA-DATA>
sudo cat /etc/kubernetes/pki/ca.crt | base64 | tr -d '\n'
```
So you do take the raw .crt and base64 encode it again, and then remove newlines.

That would seem that we are incorrectly handling the content.

Reply:
100% correct

Reply:
The kubernetes client unpacks the base64 for us. So we haven't been storing bad data luckily.